collectors/ebpf_process.plugin/ebpf_process.c

```diff
@@ -165,35 +165,43 @@ static void int_exit(int sig)
             for ( i=getdtablesize(); i>=0; --i)
                 close(i);
 
-            int fd = open("/dev/null",O_RDWR, 0);
+            int fd = open("/dev/null", O_RDWR, 0);
             if (fd != -1) {
                 dup2 (fd, STDIN_FILENO);
                 dup2 (fd, STDOUT_FILENO);
                 dup2 (fd, STDERR_FILENO);
+
             }
 
             if (fd > 2)
-                close (fd);
+                close(fd);
 
             int sid = setsid();
             if(sid >= 0) {
                 sleep(1);
                 if(debug_log) {
                     open_developer_log();
                 }
+
                 debug(D_EXIT, "Wait for father %d die", event_pid);
                 clean_kprobe_events(developer_log, event_pid, collector_events);
+
             } else {
                 error("Cannot become session id leader, so I won't try to clean kprobe_events.\n");
             }
+
+            if (developer_log) {
+                fclose(developer_log);
+                developer_log = NULL;
+            }
+
+            for ( i=getdtablesize(); i>=0; --i)
+                close(i);
```

Comment:
We don't need to close the file descriptors immediately before exiting. Or do we?

Reply:
The warning is happening because they are not closed.
Other people had the same problem as you can see here https://github.com/haproxy/haproxy/issues/421